src/quicr_server.cpp

```diff
@@ -243,21 +296,28 @@ QuicRServer::TransportDelegate::on_recv_notify(
   const qtransport::TransportContextId& context_id,
   const qtransport::MediaStreamId& mStreamId)
 {
+
+  // TODO: Consider running this in a task/async thread
   while (true) {
     auto data = server.transport->dequeue(context_id, mStreamId);
 
     if (data.has_value()) {
-
-      // TODO: Extracting type will change when the message is encoded correctly
       uint8_t msg_type = data.value().back();
 
       messages::MessageBuffer msg_buffer{ data.value() };
-      if (msg_type == static_cast<uint8_t>(messages::MessageType::Subscribe)) {
-        server.handle_subscribe(context_id, mStreamId, std::move(msg_buffer));
 
-      } else if (msg_type ==
-                 static_cast<uint8_t>(messages::MessageType::Publish)) {
-        server.handle_publish(context_id, mStreamId, std::move(msg_buffer));
+      switch (msg_type) {
+        case static_cast<uint8_t>(messages::MessageType::Subscribe):
```

Comment:
Is this cast strictly necessary?  If msg_type is based on a numeric type, it ought not be required, I think.

Reply:
Updated.